File: src/main/java/cz/gug/newsletter/Configuration.java
package cz.gug.newsletter;

import java.io.IOException;
import java.util.Locale;
import java.util.Properties;
import java.util.logging.Level;
import java.util.logging.Logger;

import org.joda.time.DateTimeZone;

public class Configuration {

	public static final String[] MONTH_NAMES = {"lednu", "únoru", "březnu", "dubnu", "květnu", "červnu", "červenci", "srpnu",
			"září", "říjnu", "listopadu", "prosinci"};

	private static final Logger LOG = Logger.getLogger(Configuration.class.getName());

	private static final String PROPERTIES_FILE = "/newsletter.properties";
	private static Configuration INSTANCE;

	private Properties properties;

	static {
		try {
			INSTANCE = new Configuration();
		} catch (IOException e) {
			LOG.log(Level.SEVERE, "Failed to read config file.", e);
		}
	}

	private Configuration() throws IOException {
		properties = new Properties();
		properties.load(Configuration.class.getResourceAsStream(PROPERTIES_FILE));
	}

	public static synchronized Configuration getInstance() {
		return INSTANCE;
	}

	public String getProperty(String key) {
		return properties.getProperty(key);
	}

	public Locale getLocale() {
		return Locale.forLanguageTag(getProperty("locale"));
	}

	public DateTimeZone getTimeZone() {
		return DateTimeZone.forID(getProperty("timezone"));
	}


}

File: src/main/kotlin/cz/gug/newsletter/Configuration.kt
package cz.gug.newsletter

import org.joda.time.DateTimeZone
import java.util.*

private const val PROPERTIES_FILE = "/newsletter.properties"

object Configuration {

    private val properties: Properties = Properties().apply {
        load(Configuration::class.java.getResourceAsStream(PROPERTIES_FILE))
    }

    val locale: Locale = Locale.forLanguageTag(getProperty("locale"))

    val timeZone: DateTimeZone = DateTimeZone.forID(getProperty("timezone"))

    val MONTH_NAMES = listOf(
        "lednu",
        "únoru",
        "březnu",
        "dubnu",
        "květnu",
        "červnu",
        "červenci",
        "srpnu",
        "září",
        "říjnu",
        "listopadu",
        "prosinci"
    )

    fun getProperty(key: String): String {
        return properties.getProperty(key)
    }
}
